klippy/extras/probe.py

```diff
@@ -40,26 +40,50 @@ def __init__(self, config, probe, query_endstop=None):
         self.printer = config.get_printer()
         self.probe = probe
         self.query_endstop = query_endstop
-        self.name = config.get_name()
+        name = config.get_name().split(' ')
+        self.name       = name[0]
+        self.probe_name = name[1] if len(name) > 1 else 'probe'
         gcode = self.printer.lookup_object('gcode')
+        # modify probe commands if named probes are used
+        query_probe_cmd          = 'QUERY_PROBE'
+        probe_cmd                = 'PROBE'
+        probe_calibrate_cmd      = 'PROBE_CALIBRATE'
+        probe_accuracy_cmd       = 'PROBE_ACCURACY'
+        z_offset_apply_probe_cmd = 'Z_OFFSET_APPLY_PROBE'
+        if self.probe_name != 'probe':
+            query_probe_cmd          += '_' + self.probe_name.upper()
+            probe_cmd                += '_' + self.probe_name.upper()
+            probe_calibrate_cmd      += '_' + self.probe_name.upper()
+            probe_accuracy_cmd       += '_' + self.probe_name.upper()
+            z_offset_apply_probe_cmd += '_' + self.probe_name.upper()
```

Comment:
This is a bit of an odd pattern to use - it would make much more sense to specify which probe to use as an input into the command. For instance, `QUERY_PROBE` would become `QUERY_PROBE PROBE=<name>`, with a default name for the backwards compatibility case.

Reply:
The functionality you describe is already there...  

Any of the existing commands can be run as you describe in the existing edited version of probe.py:
PROBE PROBE=pinda0
QUERY_PROBE PROBE=pinda0
PROBE_ACCURACY PROBE=pinda0
and so on...

If no "PROBE=" is provided it defaults to "probe" by default.

The added functionality you are highlighting is that of re-naming commands which primarily serves to avoid causing an error by registering duplicate macros.

These renamed macros end up being duplicate functions.  The following will perform the same action:
PROBE PROBE=pinda0
OR
PROBE_PINDA0

So, this could be removed by blocking the registration of these macros for any but the default probe.